forgerock-openbanking-uk-aspsp-rs/forgerock-openbanking-uk-aspsp-rs-mock-store/forgerock-openbanking-uk-aspsp-rs-mock-store-server/src/main/java/com/forgerock/openbanking/aspsp/rs/store/api/openbanking/payment/v3_1/file/FilePaymentsApiController.java

```diff
@@ -225,9 +225,13 @@ public ResponseEntity getFilePaymentsFilePaymentIdReportFile(
                                         .toOBError1(filePaymentId))
                 );
         log.debug("Consent '{}' exists so generating a report file for type: '{}'", consent.getId(), consent.getStatus(), consent.getFileType());
-        final String reportFile = paymentReportFileService.createPaymentReport(consent);
-        log.debug("Generated report file for consent: '{}'", consent.getId());
-        return ResponseEntity.ok(reportFile);
+        try {
+            final String reportFile = paymentReportFileService.createPaymentReport(consent);
+            log.debug("Generated report file for consent: '{}'", consent.getId());
+            return ResponseEntity.ok(reportFile);
+        } catch (IllegalArgumentException exception) {
```

Comment:
IAE is a bit vague and could be thrown by other parts of the code in `createPaymentReport()` (like the `checkArgument` precondition).

Reply:
Done